fiatconnect-api.md

```diff
@@ -1710,14 +1713,78 @@ This endpoint MUST fail when the user has no transfer on file with the provided
+
+## 4.1. Transfers In
+
+Statuses for transfers in must progress through the state machine shown below, ultimately ending at one of the two *terminal* states:
+`TransferFailed` or `TransferComplete`. A server SHOULD make a best effort to reach a terminal transfer state within the `settlementTimeUpperBound`
+specified for the quote associated with the transfer, if one exists. Note that since a transfer in may fail after the server has received fiat funds
+from the user's fiat account but before crypto funds are sent to the user, the server MUST return any fiat funds to the user's fiat account in case
+of such a failure.
+
+```mermaid
+graph LR
+   TransferStarted---->TransferPending
+   TransferPending---->TransferFiatFundsDebited
+   TransferFiatFundsDebited---->TransferWaitingForFiatFunds
+   TransferWaitingForFiatFunds---->TransferReceivedFiatFunds
+   TransferReceivedFiatFunds---->TransferSendingCryptoFunds
+   TransferSendingCryptoFunds---->TransferComplete
+
+   TransferStarted---->TransferFailed
+   TransferPending---->TransferFailed
+   TransferFiatFundsDebited---->TransferFailed
+   TransferWaitingForFiatFunds---->TransferFailed
+   TransferReceivedFiatFunds---->TransferFailed
+   TransferSendingCryptoFunds---->TransferFailed
+```
+
+## 4.2. Transfers Out
+
+Statuses for transfers out must progress through the state machine shown below, ultimately ending at one of the two *terminal* states:
+`TransferFailed` or `TransferComplete`. Note that at the beginning of a transfer out, a server may perform AML checks. If these AML checks
+fail, the server SHOULD progress the state machine to `TransferAmlFailed` before moving to the terminal `TransferFailed` status. A server SHOULD
+make a best effort to reach a terminal transfer state within the `settlementTimeUpperBound` specified for the quote associated with the transfer,
+if one exists.
+
+Transfers out are fundamentally different from transfer in in that they require user interaction partway through the state machine. Once a
+transfer out's state moves to `TransferReadyForUserToSendCryptoFunds`, the server should wait for the user to send the proper amount of crypto
+funds to the address specified for the transfer. Once the user has sent the proper amount of funds, the server SHOULD progress the state to
+`TransferReceivedCryptoFunds`, and continue with the transfer out.
+
+Like with transfers in, a transfer out may fail after the server has received the user's crypto funds, but before the server has credited
+their fiat account with fiat funds. In case of such a failure, the server MUST return any crypto funds sent by the user back to the user's
+crypto address.
+
+```mermaid
+graph LR
+   TransferStarted---->TransferPending
+   TransferPending---->TransferReadyForUserToSendCryptoFunds
+   TransferReadyForUserToSendCryptoFunds---->TransferReceivedCryptoFunds
+   TransferReceivedCryptoFunds---->TransferFiatFundsCredited
+   TransferFiatFundsCredited---->TransferComplete
```

Comment:
could these be one state? (TransferFiatFundsCredited and TransferComplete)

I'm noticing the transfer can't fail after TransferFiatFundsCredited...